Language and grammar changes (#6080)
medic/cht-core: config/default/GUIDE.md

> # Configuring Reference Applications

Reviewer: It is very specifically and intentionally a "cookbook". Ie it is a collection of "recipes": specific actions to solving specific problems.

This distinguishes it from a general guide / api documentation that is more thorough.

Reviewer 2: Makes sense, I think the intention was to change the "configuration" language more than the "cookbook" portion. @CescaGarrett feel free to comment if I am mistaken.

```suggestion
# Cookbook for Reference Applications
```

> These instructions are aimed at developers and technical people who are exploring our Reference Application and want to make small changes to see what is possible.

Reviewer 2: ```suggestion
These instructions are aimed at developers and technical people who are exploring the Reference Application and want to make small changes to see what is possible.
```

## Modifying Forms

> Generally, the forms are created and edited in XLSX format (a spreadsheet, which is openable with Excel, Libreoffice or your favourite office tool) and later converted into XML format to upload to the instance. The forms are categorized into app forms (Action / Task forms such as pregnancy registration, delivery report) and contact forms (Place/Person forms).

Reviewer 2: ```suggestion
Generally, the forms are created and edited in XLSX format (a spreadsheet format that can be opened with Excel, Libreoffice, Google Sheets, or your favourite Office tool) and later converted into a XML format to upload to the instance. The forms are categorized into app forms (Action or Task forms such as pregnancy registration, delivery report) and contact forms (Place or Person forms).
```

> Generally, the forms are created and edited in XLSX format (a spreadsheet format that can be opened with Excel, Libreoffice, Google Sheets, or your favourite Office tool) and later converted into a XML format to upload to the instance. The forms are categorized into app forms (Action or Task forms such as pregnancy registration, delivery report) and contact forms (Place or Person forms).

Reviewer: Wouldn't "an XML" would be correct, since it's not how it's spelt but how it sounds ("ex-em-el").

I'm uncomfortable capitalising and splitting into "Action or Task" is not correct, since capitalisation implies that it's a proper noun or otherwise named thing, which isn't really the case and hasn't been used elsewhere that I'm aware of.

Reviewer 2: Will update to "an XML".

As for the Action or Task form, it is the language used in docs.communityhealthtoolkit.org's [ANC Reference App Overview](https://docs.communityhealthtoolkit.org/resource-overviews/anc-reference-app-overview.pdf) (see the Form Hierarchy slide) so think it is fine even though it isn't codified in the app. I also felt uncomfortable with it at first since it wasn't a true distinction in the app, but since app designers have been referencing forms as such I am supportive of using it here. I will change to lower case for now and we can revise later if needed.

Reviewer 2: ```suggestion
Generally, each form is created and edited as an [XLSForm](http://xlsform.org) file, which is an Excel spreadsheet file that uses a special notation, and can also be opened with Libreoffice, Google Sheets, or your favourite Office tool. The XLSForm is later converted into an [XForm](https://opendatakit.github.io/xforms-spec/) XML file to upload to the instance. The forms are categorized into app forms (action or task forms such as pregnancy registration, delivery report) and contact forms (Place or Person forms).
```

> Generally, the forms are created and edited in XLSX format (a spreadsheet format that can be opened with Excel, Libreoffice, Google Sheets, or your favourite Office tool) and later converted into a XML format to upload to the instance. The forms are categorized into app forms (Action or Task forms such as pregnancy registration, delivery report) and contact forms (Place or Person forms).

Reviewer 2: ```suggestion
Generally, each form is created and edited as an [XLSForm](http://xlsform.org) file, which is an Excel spreadsheet file that uses a special notation, and can also be opened with Libreoffice, Google Sheets, or your favourite Office tool. The XLSForm is later converted into an [XForm](https://opendatakit.github.io/xforms-spec/) XML file to upload to the instance. The forms are categorized into app forms (action or task forms such as pregnancy registration, delivery report) and contact forms (Place or Person forms).
```

### Adding a new page of questions

> To add a new page, you need to add a group. The group starts with a row that has type “begin group” and ends with type “end group”.  The "name" field is also required for "begin group" type. The "name" field is not required in the “end group” row, but can help provide clarity in nested groups.

Reviewer 2: ```suggestion
To add a new page, you need to add a group. The group starts with a row that has type “begin group” and ends with type “end group”.  The "name" field for the "begin group" row is used as the page title. The "name" field in the “end group” row is not shown in the form, but can help provide clarity for large forms and those with nested groups.
```

Tasks are defined inside the file `tasks.js`.

A Task can apply to certain contacts or contacts with specific reports with specific properties that are defined in `appliesIf` function of the task. The criteria for resolving/clearing the Task is defined in the `resolvedIf` function.

> To show up a Task following conditions need to be satisfied:

Reviewer 2: ```suggestion
To show up, a Task following conditions need to be satisfied:
```

> To show up a Task following conditions need to be satisfied:

Reviewer 2: ```suggestion
The following conditions need to be satisfied for a Task to show:
```
